### PR TITLE
Cleanup sync createWorld mess

### PR DESCRIPTION
**Pull request by Pablete1234**
As suggested by a commit message, the CountDownLatch is a cleaner way to do this type of syncronization

### CODE REVIEW

File: src/main/java/tc/oc/pgm/match/MatchManagerImpl.java [hunk 1]
```diff
@@ -197,39 +197,30 @@ private World createMatchWorld(String worldName, MapInfo info, TerrainModule ter
...
-      synchronized (LOCK) {
-        try {
-          LOCK.wait();
-        } catch (InterruptedException ignore) {
-        }
-      }
+    try {
+      latch.await();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/match/MatchManagerImpl.java]: Logger?

Comment (Pablete1234) [reply src/main/java/tc/oc/pgm/match/MatchManagerImpl.java]: This is the main thread, if it's getting interrupted it's likely shutting down the server anyways, but will add the logger

Comment (Electroid) [reply src/main/java/tc/oc/pgm/match/MatchManagerImpl.java]: `logger.log(Level.SEVERE, "...", e)`

We want to always use the logger for when I add Sentry/other logging support.

Comment (Pablete1234) [reply src/main/java/tc/oc/pgm/match/MatchManagerImpl.java]: Had already changed it, was failing to push stuff properly

**Pull request merged**